comp/metadata/packagesigning/utils/rpm.go

```diff
@@ -0,0 +1,179 @@
+)
+
+// Not const for testing purpose
+var (
+	YumConf = "/etc/yum.conf"
+	YumRepo = "/etc/yum.repos.d/"
+)
+
+const (
+	dnfConf  = "/etc/dnf/dnf.conf"
+	zyppConf = "/etc/zypp/zypp.conf"
+	zyppRepo = "/etc/zypp/repos.d/"
+)
+
+// getMainGPGCheck returns gpgcheck and repo_gpgcheck setting for [main] table
+func getMainGPGCheck(pkgManager string) (bool, bool) {
+	repoConfig, _ := GetRepoPathFromPkgManager(pkgManager)
+	if repoConfig == "" {
+		// if we end up in a non supported distribution
+		return false, false
+	}
+	defaultValue := strings.Contains(repoConfig, "zypp") // Settings are enabled by default on SUSE, disabled otherwise
+	mainConf, _ := ParseRPMRepoFile(repoConfig, MainData{Gpgcheck: defaultValue, LocalpkgGpgcheck: defaultValue, RepoGpgcheck: defaultValue})
+	return mainConf.Gpgcheck, mainConf.RepoGpgcheck
+}
+
+// GetRepoPathFromPkgManager returns the path to the configuration file and the path to the repository files for RH or SUSE based OS
+func GetRepoPathFromPkgManager(pkgManager string) (string, string) {
+	if pkgManager == "yum" {
+		return YumConf, YumRepo
+	} else if pkgManager == "dnf" {
+		return dnfConf, YumRepo
+	} else if pkgManager == "zypper" {
+		return zyppConf, zyppRepo
+	}
+	return "", ""
+}
+
+// Repository is a struct to store the repo name
+type Repository struct {
+	Name         string `json:"name"`
+	Enabled      bool   `json:"enabled"`
+	GPGCheck     bool   `json:"gpgcheck"`
+	RepoGPGCheck bool   `json:"repo_gpgcheck"`
+}
+
+// MainData contains the global definitions of gpg checks
+type MainData struct {
+	Gpgcheck         bool
+	LocalpkgGpgcheck bool
+	RepoGpgcheck     bool
+}
+type repoData struct {
+	baseurl      []string
+	enabled      bool
+	metalink     string
+	mirrorlist   string
+	gpgcheck     bool
+	repoGpgcheck bool
+	gpgkey       []string
+}
+
+type multiLine struct {
+	inside bool
+	name   string
+}
+
+// ParseRPMRepoFile extracts information from yum repo files
+// Save the global gpgcheck value when encountering a [main] table (should only occur on `/etc/yum.conf`)
+// Match several entries in gpgkey field, either file references (file://) or http(s)://. From observations,
+// these reference can be separated either by space or by new line. We assume it possible to mix file and http references
+func ParseRPMRepoFile(inputFile string, mainConf MainData) (MainData, map[string][]Repository) {
+	main := mainConf
+	file, err := os.Open(inputFile)
+	if err != nil {
+		return main, nil
+	}
+	defer file.Close()
+	defaultValue := strings.Contains(inputFile, "zypp") // Settings are enabled by default on SUSE, disabled otherwise
+	reposPerKey := make(map[string][]Repository)
+	table := regexp.MustCompile(`\[[A-Za-z0-9_\-\.\/ ]+\]`)
+	field := regexp.MustCompile(`^([a-z_]+)\s?=\s?(.*)$`)
+	nextLine := multiLine{inside: false, name: ""}
+	repo := repoData{enabled: true, gpgcheck: defaultValue, repoGpgcheck: defaultValue}
+	var repos []repoData
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
```

Comment:
I don't see much error handling here. Is it ok to ignore all of them (my guess is that it's not)?

```
a client may not assume that no error occurred. Instead it must check the scanner's ErrorCount or the number of calls of the error handler
```

Same question around handling of badly formatted files in this and other impls (apt/etc).

Reply:
Correct the management of `scanner.Err()` vanished, let's be more accurate